internal/scorecard/alpha/tests/olm.go

```diff
@@ -83,3 +113,73 @@ func StatusDescriptorsTest(bundle registry.Bundle) scapiv1alpha2.ScorecardTestRe
 	r.Suggestions = make([]string, 0)
 	return r
 }
+
+func registryToAPICSV(csv *registry.ClusterServiceVersion) (*operators.ClusterServiceVersion, error) {
+	var apiCSV operators.ClusterServiceVersion
+	csvBytes, err := json.Marshal(csv)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(csvBytes, &apiCSV)
+	if err != nil {
+		return nil, err
+	}
+	return &apiCSV, nil
+}
+
+// TODO This is the validation we did in v1, but it looks like it only validates fields that
+// are in the example CRs, if you have a field in your CRD that isn't present in one of your examples,
+// I don't think it will be validated.
+func checkOwnedCSVDescriptors(cr unstructured.Unstructured, csv *operators.ClusterServiceVersion,
+	descriptor string, r scapiv1alpha2.ScorecardTestResult) scapiv1alpha2.ScorecardTestResult {
+
+	if cr.Object[descriptor] == nil {
+		r.State = scapiv1alpha2.FailState
+		return r
+	}
+
+	block := cr.Object[descriptor].(map[string]interface{})
+
+	var crd *operators.CRDDescription
+	for _, owned := range csv.Spec.CustomResourceDefinitions.Owned {
+		if owned.Kind == cr.GetKind() {
+			crd = &owned
+			break
+		}
+	}
+
+	if crd == nil {
+		msg := fmt.Sprintf("Failed to find an owned CRD for CR %s with GVK %s", cr.GetName(), cr.GroupVersionKind().String())
+		r.Errors = append(r.Errors, msg)
+		r.State = scapiv1alpha2.FailState
+		return r
+	}
+
+	if descriptor == statusDescriptor {
+		for key := range block {
+			for _, statDesc := range crd.StatusDescriptors {
+				if statDesc.Path == key {
+					delete(block, key)
+					break
+				}
+			}
+		}
+	}
+	if descriptor == specDescriptor {
+		for key := range block {
+			for _, specDesc := range crd.SpecDescriptors {
+				if specDesc.Path == key {
+					delete(block, key)
+					break
+				}
+			}
+		}
+	}
+
+	for key := range block {
+		r.Errors = append(r.Errors, fmt.Sprintf("%s does not have a %s descriptor", key, descriptor))
+		r.Suggestions = append(r.Suggestions, fmt.Sprintf("Add a %s descriptor for %s", descriptor, key))
+		r.State = scapiv1alpha2.FailState
+	}
+	return r
```

Comment:
Are these changes covered by tests?

Reply:
partially, though I'll note that these are not changes, this logic is directly pulled from the existing test. Currently the testdata we're using is pretty sparse and focuses on the happy path (for all the olm/basic tests), as a followup task we should definitely improve the quality of the available testdata so that the failure cases are more thoroughly exercised.